server/app/controllers/admin/AdminProgramStatusesController.java

```diff
@@ -103,11 +123,17 @@ public Result createOrUpdate(Http.Request request, long programId)
     }
     boolean isUpdate =
         previousStatusCount == mutateResult.getResult().statusDefinitions().getStatuses().size();
-    return redirect(indexUrl).flashing("success", isUpdate ? "Status updated" : "Status created");
+    String toastMessage = isUpdate ? "Status updated" : "Status created";
+    if (setDefault
+        && previousDefaultStatus.isPresent()
+        && !previousDefaultStatus.get().statusText().equals(form.get().getConfiguredStatusText())) {
```

Comment:
Changing this a little bit, because it complains about `form.get().getConfiguredStatusText()` not being final when used inside a lambda, and because I think I actually want it to show the "updated" message when you are setting a default when there was no default previously.

```
    final String configuredStatusText = form.get().getConfiguredStatusText();
    if (setDefault && previousDefaultStatus.map(
        status -> 
          !status.statusText().equals(configuredStatusText)).orElse(true)) {
```

Reply:
LGTM, thanks!